aixplain/modules/agent/__init__.py

```diff
@@ -165,19 +164,43 @@ def run(
                 max_iterations=max_iterations,
                 output_format=output_format,
             )
-            if response["status"] == "FAILED":
+            if response["status"] == ResponseStatus.FAILED:
                 end = time.time()
                 response["elapsed_time"] = end - start
                 return response
             poll_url = response["url"]
             end = time.time()
-            response = self.sync_poll(poll_url, name=name, timeout=timeout, wait_time=wait_time)
-            return response
+            result = self.sync_poll(poll_url, name=name, timeout=timeout, wait_time=wait_time)
+            result_data = result.data
+            return AgentResponse(
+                status=ResponseStatus.SUCCESS,
+                completed=True,
+                data=AgentResponseData(
+                    input=result_data.get("input"),
+                    output=result_data.get("output"),
+                    session_id=result_data.get("session_id"),
+                    intermediate_steps=result_data.get("intermediateSteps"),
+                    execution_stats=result_data.get("executionStats"),
+                ),
+                used_credits=result_data.get("usedCredits", 0.0),
+                run_time=result_data.get("runTime", end - start),
+            )
         except Exception as e:
             msg = f"Error in request for {name} - {traceback.format_exc()}"
             logging.error(f"Agent Run: Error in running for {name}: {e}")
             end = time.time()
-            return {"status": "FAILED", "error": msg, "elapsed_time": end - start}
+            return AgentResponse(
+                status=ResponseStatus.FAILED,
+                data=AgentResponseData(
+                    input=data,
+                    output=None,
+                    session_id=result_data.get("session_id"),
```

Comment:
There is two session ids. This is raising an exception. Please run pre-commit to avoid these errors!